src/lib/sw_engine/tvgSwRenderer.cpp

```diff
@@ -143,11 +143,21 @@ struct SwShapeTask : SwTask
             if (shape.rle) {
                 if (clipper->rect) rleClipRect(shape.rle, &clipper->bbox);
                 else if (clipper->rle) rleClipPath(shape.rle, clipper->rle);
+                else {
```

Comment:
this looks not good to access internal rle data in renderer.
probably.... -> goto err?
also, in this case it's not meaning to iterate for loop, you can break it immediately.